src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaInductionPersistenceAdapter.kt

```diff
@@ -89,8 +90,16 @@ class JpaInductionPersistenceAdapter(
       return previousQualificationsRepository.saveAndFlush(previousQualificationsEntity)
     } else {
       // Prisoner does not already have previous qualifications, and the DTO has qualifications. We need to create them
+      val createPreviousQualificationsDto = with(updateInductionDto.previousQualifications!!) {
+        CreatePreviousQualificationsDto(
+          prisonNumber = prisonNumber,
+          educationLevel = educationLevel!!,
+          qualifications = qualifications,
+          prisonId = prisonId,
+        )
+      }
       return previousQualificationsRepository.saveAndFlush(
-        previousQualificationsMapper.fromUpdateDtoToNewEntity(updateInductionDto.previousQualifications)!!,
+        previousQualificationsMapper.fromCreateDtoToEntity(createPreviousQualificationsDto)!!,
```

Comment:
This is the core fix.
Previously we had a mapper that took a `UpdatePreviousQualificationsDto` instance and returned a new un-persisted `PreviousQualificationsEntity` instance.
At compile and unit test time this worked; but the resultant `PreviousQualificationsEntity` was missing `prisonNumber` because that field does not exist in `UpdatePreviousQualificationsDto` (it doest belong in `UpdatePreviousQualificationsDto` as it's not a property we use to reference/locate the right record, and its not a property we support updating)
At runtime this failed when JPA tried to persist the `PreviousQualificationsEntity` because `prisonNumber` is a non-nullable column.

Essentially the mapper is wrong, in that it doesn't make sense to create a new `PreviousQualificationsEntity` from an `UpdatePreviousQualificationsDto` (create a new thing from something that describes an update 🤔 👎 )

We already have a mapper that creates a new `PreviousQualificationsEntity` but from a `CreatePreviousQualificationsDto` which contains all of the required data, so my fix is to use that mapper instead, and to delete the other mapper entirely